File: src/test/starterkit/starterAI.java
import java.util.*;
import java.io.*;

class Player {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);

        // ALL CUSTOMERS INPUT: to ignore until Bronze
        int numAllCustomers = in.nextInt();
        for (int i = 0; i < numAllCustomers; i++) {
            String customerItem = in.next();
            int customerAward = in.nextInt();
        }

        // KITCHEN INPUT
        Kitchen k = new Kitchen();
        k.init(in);

        // game loop
        while (true) {
            int turnsRemaining = in.nextInt();

            // PLAYERS INPUT
            int playerX = in.nextInt();
            int playerY = in.nextInt();
            String playerItem = in.next();
            int partnerX = in.nextInt();
            int partnerY = in.nextInt();
            String partnerItem = in.next();

            int numTablesWithItems = in.nextInt();
            for (int i = 0; i < numTablesWithItems; i++) {
                int tableX = in.nextInt();
                int tableY = in.nextInt();
                String item = in.next();
            }
            // oven to ignore until bronze
            String ovenContents = in.next();
            int ovenTimer = in.nextInt();

            // CURRENT CUSTOMERS INPUT
            int numCustomers = in.nextInt();
            for (int i = 0; i < numCustomers; i++) {
                String customerItem = in.next();
                int customerAward = in.nextInt();
            }

            // GAME LOGIC
            // fectch dish, then blueberries, then ice cream
            if (!playerItem.contains("DISH")) {
                Table table = k.tables.stream().filter(t -> t.isDish()).findFirst().get();
                table.use();
            }
            else if (!playerItem.contains("BLUEBERRIES")) {
                k.tables.stream().filter(t -> t.isBlueBerries()).findFirst().get().use();
            }
            else if (!playerItem.contains("ICE_CREAM")) {
                k.tables.stream().filter(t -> t.isIceCream()).findFirst().get().use();
            }
            else {
                k.tables.stream().filter(t -> t.isWindow()).findFirst().get().use();
            }
        }
    }
}

class Kitchen {

    List<Table> tables;

    public Kitchen() {
        tables = new ArrayList<Table>();
    }

    public void init(Scanner in) {
        in.nextLine();
        for (int i = 0; i < 7; i++) {
            String kitchenLine = in.nextLine();
            for (int j = 0; j < kitchenLine.length(); j++) {
                char c = kitchenLine.charAt(j);
                TableType t = TableType.get(c);
                if (t != null) {
                    tables.add(new Table(t, j, i));
                }
            }
            System.err.println(kitchenLine);
        }
    }
}

class Position {

    int x, y;

    public Position(int x, int y) {
        this.x = x;
        this.y = y;
    }

    // @TODO
    public int distance(Position p) {
        return 0;
    }
}

class Table {

    TableType type;

    int x, y;

    public Table(TableType t, int x, int y) {
        type = t;
        this.x = x;
        this.y = y;
    }

    public void error() {
        System.err.println("Table of type "+type+" at "+x+" "+y);
    }

    public boolean isDish() {
        System.err.println(type);
        return isTableType(TableType.PLATES);
    }

    public boolean isBlueBerries() {
        return isTableType(TableType.BLUEBERRY);
    }

    public boolean isIceCream() {
        return isTableType(TableType.ICE_CREAM);
    }

    public boolean isWindow() {
        return isTableType(TableType.WINDOW);
    }

    private boolean isTableType(TableType t) {
        System.err.println(type.equals(t));
        return type.equals(t);
    }

    public void use() {
        System.out.println("USE "+x+" "+y+" my message");
    }
}

enum TableType {

    BLUEBERRY,
    ICE_CREAM,
    PLATES,
    WINDOW;

    static TableType get(Character c) {
        switch (c) {
            case 'D':
                return PLATES;
            case 'B':
                return BLUEBERRY;
            case 'I':
                return ICE_CREAM;
            case 'W':
                return WINDOW;

        }
        return null;
    }
}

File: src/test/starterkit/starterAI.py
#Begin Imports
import sys
import math
#End Imports

#Begin Util Code
def log(x):
    print(x, file=sys.stderr)

class Player:
    def __init__(self):
        self.x = 0
        self.y = 0
        self.item = NONE

class Tile:
    def __init__(self, x, y, name):
        self.x = x
        self.y = y
        self.name = name
        self.item = None
<<<<<<< HEAD
    
=======

>>>>>>> 066833a... update python starter
    def parse_name(self):
        return self.name.split("-")

    def __repr__(self):
        return "Tile: " + str(self.x) + ", " + str(self.y)

# Cells
BLUEBERRIES_CRATE = "B"
ICE_CREAM_CRATE = "I"
WINDOW = "W"
EMPTY_TABLE = "#"
DISHWASHER = "D"
FLOOR_CELL = "."

# Items
NONE = "NONE"
DISH = "DISH"
ICE_CREAM = "ICE_CREAM"
BLUEBERRIES = "BLUEBERRIES"

class Game:
    def __init__(self):
        self.player = Player()
        self.partner = Player()
        self.tiles = []

    def addTile(self, x, y, tileChar):
        if tileChar != '.':
            self.tiles.append(Tile(x, y, tileChar))

    def getTileByName(self, name):
        for t in self.tiles:
            if t.name == name:
                return t

        #If tile not found
        log("Error: Tile not found in function getTileByName")
<<<<<<< HEAD
        
=======

>>>>>>> 066833a... update python starter
    def getTileByItem(self, item):
        for t in self.tiles:
            if t.item == item:
                return t
<<<<<<< HEAD
        
        #If tile not found
        log("Error: Tile not found in function getTileByItem")
    
=======

        #If tile not found
        log("Error: Tile not found in function getTileByItem")

>>>>>>> 066833a... update python starter
    def getTileByCoords(self, x, y):
        for t in self.tiles:
            if t.x == x and t.y == y:
                return t

        #If tile not found
        log("Error: Tile not found in function getTileByCoords")

    def updatePlayer(self, x, y, item):
        self.player.x = x
        self.player.y = y
        self.player.item = item

    def updatePartner(self, x, y, item):
        self.partner.x = x
        self.partner.y = y
        self.partner.item = item

    def use(self, tile):
<<<<<<< HEAD
        print("USE", tile.x, tile.y,"; Python Starter AI")
    
=======
        print("USE", tile.x, tile.y)

>>>>>>> 066833a... update python starter
    def move(self, tile):
        print("MOVE", tile.x, tile.y)
#End Util code

#Begin game code
game = Game()

# ALL CUSTOMERS INPUT: to ignore until bronze
num_all_customers = int(input())
for i in range(num_all_customers):
    # customer_item: the food the customer is waiting for
    # customer_award: the number of points awarded for delivering the food
    customer_item, customer_award = input().split()
    customer_award = int(customer_award)

# KITCHEN INPUT
for y in range(7):
    kitchen_line = input()
    for x, tileChar in enumerate(kitchen_line):
        game.addTile(x, y, tileChar)

# game loop
while True:
    turns_remaining = int(input())

    # PLAYERS INPUT
    #Gather and update player information
    player_x, player_y, player_item = input().split()
    player_x = int(player_x)
    player_y = int(player_y)
    game.updatePlayer(player_x, player_y, player_item)

    #Gather and update partner information
    partner_x, partner_y, partner_item = input().split()
    partner_x = int(partner_x)
    partner_y = int(partner_y)
    game.updatePartner(partner_x, partner_y, partner_item)

    #Gather and update table information
    for t in game.tiles:        
        t.item = None
    num_tables_with_items = int(input())  # the number of tables in the kitchen that currently hold an item
    for i in range(num_tables_with_items):
        table_x, table_y, item = input().split()
        table_x = int(table_x)
        table_y = int(table_y)
        game.getTileByCoords(table_x, table_y).item = item

    # oven_contents: ignore until bronze league
    oven_contents, oven_timer = input().split()
    oven_timer = int(oven_timer)
    num_customers = int(input())  # the number of customers currently waiting for food
    for i in range(num_customers):
        customer_item, customer_award = input().split()
        customer_award = int(customer_award)

    # GAME LOGIC
    #Gather plate and Icecream
    if DISH not in game.player.item:
        game.use(game.getTileByName(DISHWASHER))
    elif ICE_CREAM not in game.player.item:
        game.use(game.getTileByName(ICE_CREAM_CRATE))
    else:
        game.use(game.getTileByName(EMPTY_TABLE))
#End game code